sphinx_gallery/gen_rst.py

```diff
@@ -292,18 +295,22 @@ def check_md5sum_change(src_file):
         if src_md5 == ref_md5:
             src_file_changed = False
 
-    if src_file_changed:
-        with open(src_md5_file, 'w') as file_checksum:
-            file_checksum.write(src_md5)
-
     return src_file_changed
 
 
-def _plots_are_current(src_file, image_path):
+def _plots_are_current(src_file, image_path_template):
     """Test existence of image file and no change in md5sum of
     example"""
 
-    has_image = os.path.exists(image_path)
+    has_image = os.path.exists(image_path_template.format(1))
```

Comment:
I think we discussed that last time that hashes should only be written once the example succeeds. This way we don't need this slightly peculiar logic of checking whether an image has been generated.